pkg/build/cmd/whatsnewchecker_test.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/build/config"
+	"github.com/stretchr/testify/require"
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	DroneBuildEvent = "DRONE_BUILD_EVENT"
+	DroneTag        = "DRONE_TAG"
+)
+
+const whatsNewUrl = "https://grafana.com/docs/grafana/next/whatsnew/whats-new-in-"
+
+func TestWhatsNewChecker(t *testing.T) {
+	tests := []struct {
+		envMap             map[string]string
+		packageJsonVersion string
+		name               string
+		wantErr            bool
+		errMsg             string
+	}{
+		{envMap: map[string]string{DroneBuildEvent: config.PullRequest}, packageJsonVersion: "", name: "non-tag event", wantErr: true, errMsg: "non-tag pipeline, exiting"},
+		{envMap: map[string]string{DroneBuildEvent: config.Tag, DroneTag: "abcd123"}, packageJsonVersion: "", name: "non-semver compatible", wantErr: true, errMsg: "non-semver compatible version vabcd123, exiting"},
+		{envMap: map[string]string{DroneBuildEvent: config.Tag, DroneTag: "10.0.0"}, packageJsonVersion: "v10-0", name: "package.json version matches tag", wantErr: false},
+		{envMap: map[string]string{DroneBuildEvent: config.Tag, DroneTag: "10.0.0"}, packageJsonVersion: "v9-5", name: "package.json doesn't match tag", wantErr: true, errMsg: "whatsNewUrl in package.json needs to be updated to https://grafana.com/docs/grafana/next/whatsnew/whats-new-in-v10-0/"},
+	}
+	for _, tt := range tests {
+		app := cli.NewApp()
+		app.Version = "1.0.0"
+		context := cli.NewContext(app, &flag.FlagSet{}, nil)
+		t.Run(tt.name, func(t *testing.T) {
+			setUpEnv(t, tt.envMap)
+			err := createTempPackageJson(t, tt.packageJsonVersion)
+			require.NoError(t, err)
+			defer deleteTempPackageJson(t)
```

Comment:
You could also move that `deleteTempPackageJson` directly into the `createTempPackageJson` helper using `t.Cleanup` 🙂

Reply:
True, thanks! Updated!